distributed/worker.py

```diff
@@ -3543,7 +3627,31 @@ def get_call_stack(self, comm=None, keys=None):
     def _notify_plugins(self, method_name, *args, **kwargs):
         for name, plugin in self.plugins.items():
             if hasattr(plugin, method_name):
-                if method_name == "release_key":
+                if (
+                    not getattr(plugin, "enum_task_state_names", False)
+                    and hasattr(self, "release_key")
+                    and method_name == "transition"
+                ):
+                    warnings.warn(
+                        "The `WorkerPlugin.transition` hook will change soon "
+                        "and will no longer receive string values for start "
+                        "finish but instead an instance of "
+                        f"`dask.distributed.worker.{WTSName.__name__}`. "
+                        "To subscribe to the new behaviour already, set "
+                        "`WorkerPlugin.enum_task_state_names = True` "
+                        "for your plugin",
+                        DeprecationWarning,
+                    )
+                    args = tuple(
+                        arg.name if isinstance(arg, WorkerTaskStateName) else arg
+                        for arg in args
+                    )
+                    kwargs = {
+                        k: v.name if isinstance(v, WorkerTaskStateName) else v
+                        for k, v in kwargs.items()
+                    }
```

Comment:
```suggestion
                    args = tuple(
                        arg.value if isinstance(arg, WTSName) else arg
                        for arg in args
                    )
                    kwargs = {
                        k: v.value if isinstance(v, WTSName) else v
                        for k, v in kwargs.items()
                    }
```
for long-running, name != value